[QPPA-920] Styling navigation (Part 3 of 3) (but there will probably be a bonus round)

abarciauskas-bgse opened the pull request:
Adds dropdown navigation, styling to match the mockups and some responsive behavior.

[QPPA-920](https://jira.cms.gov/browse/QPPA-920)

Detail:
There is a with more detail on this work [tech spec](https://docs.google.com/document/d/1CL-CXHcIJifPJYUkIQQaBN15nRuDl5vSC_f3CEJv0wc/edit#). This is the third PR of 3 implement the new navigation and styling on all pages. This PR adds dropdown navigation when the page is <544px, addresses some additional styling requirements like putting content in white boxes and using the Rubik font-family, and some other minor responsive behavior tweaks.

There will be a follow up PR to improve responsive behavior, specifically to make font-sizing consistent and consistently responsive. There are additional TODOs I'm tracking [at the bottom of the tech spec](https://docs.google.com/document/d/1CL-CXHcIJifPJYUkIQQaBN15nRuDl5vSC_f3CEJv0wc/edit#heading=h.rv59pvnd8i1d)

Test plan:
* [x] adhoc testing via UI
* [x] updated unit tests

![aug-24-2017 18-05-08](https://user-images.githubusercontent.com/15016780/29694108-ebaccbf6-88f6-11e7-99f3-764f714fe507.gif)

abarciauskas-bgse commented on src/styles/app.css:
```diff
@@ -34,6 +36,16 @@ table td {
+}
+
+.inset-content {
+  padding: 1.8em;
```
No longer being used, need to remove this thanks for @samskeller

sawyerh commented on src/components/app.js:
```diff
@@ -39,24 +34,32 @@ class App extends React.Component {
-              <div className='ds-u-float--left ds-u-padding-right--6 ds-u-padding-top--2 ds-l-col--3'>
-                <ul className='ds-c-vertical-nav__subnav'>
+
+          <div className='content-container ds-l-row'>
```
`content-container` is fine — but just pointing out there's also [`ds-l-container`](https://design.cms.gov/layout/grid/) which applies similar constraints (but it would need applied to a parent of `ds-l-row`)

sawyerh commented on src/styles/app.css:
```diff
@@ -63,5 +77,40 @@ a.qpp-docs-title-back {
+  }
+}
+
+@media (min-width: 544px) {
```
As an alternative to having these style declarations that toggles an element's visibility at different breakpoints, you can accomplish the same functionality with the ["display" utility class](https://design.cms.gov/utilities/display-visibility#utilities.display-visibility.display): 

The following would hide the element on viewports larger than 544px, and show it when the viewport is smaller than 544px.

```jsx
<div className='ds-u-display--block ds-u-sm-display--none'>
```

sawyerh commented on src/components/app.js:
```diff
@@ -39,24 +34,32 @@ class App extends React.Component {
-                </div>
+            </div>
+
+            <div className='ds-u-float--left ds-u-padding-top--4 ds-l-col--12 ds-l-sm-col--9'>
```
I think you can get rid of the `ds-u-float--left` since this is a flexbox column

sawyerh commented on src/styles/left-nav.css:
```diff
@@ -0,0 +1,23 @@
+.ds-c-vertical-nav > .ds-c-vertical-nav__item {
+  border-top: none;
+  margin-bottom: 0px;
```
This should already have a `0px` margin

sawyerh commented on src/styles/left-nav.css:
```diff
@@ -0,0 +1,23 @@
+  /* QPP "Medium Teal" */
+  border-left: 4px solid #209fa6;
+  font-weight: normal;
+  color: inherit;
```
Looks like you overwrite this `color` a few lines down

abarciauskas-bgse commented:
@sawyerh Thanks! addressed all your comments in https://github.com/CMSgov/qpp-submissions-docs/pull/56/commits/c19cf6ae343e1c3bb0ca37b04b2e1f17f768ea68 and https://github.com/CMSgov/qpp-submissions-docs/pull/56/commits/754ff4c8486060c14b26fdd750013ad3aad06f22

sawyerh commented on src/components/app.js:
```diff
@@ -29,37 +29,39 @@ class App extends React.Component {
-              </form>
-            </div>
+          <div className='ds-l-container'>
+            <div className='content-container ds-l-row'>
```
Just flaggin `content-container` which looks like it's not needed anymore since the CSS rule-set got removed?

abarciauskas-bgse replied:
Good catch, updated https://github.com/CMSgov/qpp-submissions-docs/pull/56/commits/2e61099bf4e5b297c9f207f32a1408542c2844fb

kencheeto commented on src/components/left-nav.jsx:
```diff
@@ -2,22 +2,32 @@ import React from 'react';
-      <ul className="ds-c-vertical-nav__subnav">
-        <li><b>{this.props.name}</b></li>
+    return <li>
+      <ul className="ds-c-vertical-nav">
```
if we add a `left-nav` class here, we could avoid altering design system classes for left-nav specific styles (as described below)

kencheeto commented on src/styles/left-nav.css:
```diff
@@ -0,0 +1,21 @@
+  background: #f1f1f1;
+  /* QPP "Navy" */
+  color: #125496;
+}
```
these are essentially design system overrides too, but specific to the `LeftNav` component. if we add a `left-nav` class like in the comment above, these styles could be written as:

```css
.left-nav.ds-c-vertical-nav > .ds-c-vertical-nav__item {
  border-top: none;
}

.left-nav.ds-c-vertical-nav a.ds-c-vertical-nav__label:visited {
  /* QPP "Navy" */
  color: #125496;
}

.left-nav.ds-c-vertical-nav a.ds-c-vertical-nav__label {
  border: none;
}
```
and would be specific to the `LeftNav` component while still retaining the necessary specificity.

kencheeto replied:
cc @sawyerh is this a reasonable way to modify design system rules in a scoped way?

sawyerh replied:
I think a less specific selector would also work (design system selectors generally have very low specificity):

```css
.left-nav .ds-c-vertical-nav__label:visited {
  /* QPP "Navy" */
  color: #125496;
}

.left-nav .ds-c-vertical-nav__label {
  border: none;
}
```

marimiyachi commented:
omgosh this looks so good!!! 😍 great job @abarciauskas-bgse!
